azext_edge/edge/command_map.py

```diff
@@ -286,22 +287,106 @@ def load_iotops_commands(self, _):
         cmd_group.show_command("show", "show_namespace_asset")
 
     with self.command_group(
-        "iot ops ns asset create",
+        "iot ops ns asset custom",
+        command_type=namespace_resource_ops,
+    ) as cmd_group:
+        cmd_group.command("create", "create_namespace_custom_asset")
+        cmd_group.command("update", "update_namespace_custom_asset")
+
+    with self.command_group(
+        "iot ops ns asset custom dataset",
+        command_type=namespace_resource_ops,
+    ) as cmd_group:
+        cmd_group.command("add", "add_namespace_custom_asset_dataset")
+        cmd_group.command("list", "list_namespace_asset_datasets")
+        cmd_group.command("remove", "remove_namespace_asset_dataset")
+        cmd_group.show_command("show", "show_namespace_asset_dataset")
+        cmd_group.command("update", "update_namespace_custom_asset_dataset")
+
+    with self.command_group(
+        "iot ops ns asset custom dataset point",
+        command_type=namespace_resource_ops,
+    ) as cmd_group:
+        cmd_group.command("add", "add_namespace_custom_asset_dataset_point")
+        cmd_group.command("list", "list_namespace_asset_dataset_points")
+        cmd_group.command("remove", "remove_namespace_asset_dataset_point")
+
+    with self.command_group(
+        "iot ops ns asset custom event",
+        command_type=namespace_resource_ops,
+    ) as cmd_group:
+        cmd_group.command("add", "add_namespace_custom_asset_event")
+        cmd_group.command("list", "list_namespace_asset_events")
+        cmd_group.command("remove", "remove_namespace_asset_event")
+        cmd_group.show_command("show", "show_namespace_asset_event")
+        cmd_group.command("update", "update_namespace_custom_asset_event")
+
+    with self.command_group(
+        "iot ops ns asset custom event point",
+        command_type=namespace_resource_ops,
+    ) as cmd_group:
+        cmd_group.command("add", "add_namespace_custom_asset_event_point")
+        cmd_group.command("list", "list_namespace_asset_event_points")
+        cmd_group.command("remove", "remove_namespace_asset_event_point")
+
+    with self.command_group(
+        "iot ops ns asset media",
+        command_type=namespace_resource_ops,
+    ) as cmd_group:
+        cmd_group.command("create", "create_namespace_media_asset")
```

Comment:
no remove here?

Reply:
delete is `az iot ops ns asset` level since you don't need to know what "asset type" your asset is to remove/list/show

Reply:
also note that I am doing this only for the main asset commands since it is closer command level wise + the dataset/event access should depend on the asset type